back/spec/services/url_validation_service_spec.rb

```diff
@@ -0,0 +1,121 @@
+require 'rails_helper'
+
+describe UrlValidationService do
+  subject(:service) { described_class.new }
+
+  describe '#whitelisted?' do
+    it 'checks YouTube url passes through whitelist' do
```

Comment:
I've tested all of these on the combined "whitelisted" method, but let me know if it would be better to break this up over url_whitelisted and video_whitelisted specifically.